### ISSUE TITLE
Handle unsupported implementation queries

### ISSUE THREAD

**Issue by csadorf** [CasperWA/voila-optimade-client#125]
Trying to query the Crystallographic Open Database (CoD) I get the following error:
![Screen Shot 2020-09-03 at 3 55 51 PM](https://user-images.githubusercontent.com/1441208/92124365-fa4bc580-edfd-11ea-9d49-d7bef349cac5.png)

**Comment by CasperWA** [comment]
Thanks for the report @csadorf!
And yup. Known bug. I need to figure out the counter-measures for databases that do not support certain filterings.

There are indeed some ongoing efforts to improve the informational accuracy of which properties are queryable and/or supported by a database. This is being done currently in [`optimade-python-tools`](https://github.com/Materials-Consortia/optimade-python-tools).
Furthermore, I will suggest we add some general and mandatory introspective keys to the OPTIMADE specification that will help implement this degree of informational accuracy further for custom implementations (like the one for COD).

**Comment by CasperWA** [comment]
Just to be clear concerning this issue. It is _mandatory_ to support queries on this property (`nsites`) [according to the OPTIMADE specification](https://github.com/Materials-Consortia/OPTIMADE/blob/v1.0.0/optimade.rst#nsites).

So this issue is mainly related to the UX and how to deal with queries (mandatory or not) that a server implementation does not support.

**Comment by merkys** [comment]
> Just to be clear concerning this issue. It is _mandatory_ to support queries on this property (`nsites`) [according to the OPTIMADE specification](https://github.com/Materials-Consortia/OPTIMADE/blob/v1.0.0/optimade.rst#nsites).

Correct. This issue is a known bug at the COD site ([#829](https://projects.ibt.lt/repositories/issues/829)) and we are working to fix it.
